docs/src/content/docs/basic-concepts.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: Basic Concepts
+description: Understanding the core concepts of GenSX
+---
+
+GenSX is a workflow engine designed for building complex LLM applications. It's built around functional, reusable components that are composed using JSX to create and orchestrate complicated workflows.
+
+## Components
+
+Components are the building blocks of GenSX applications; they're pure TypeScript functions that:
+
+- Accept props as input
+- Return a single output or another component
+- Have zero side effects
+- Don't depend on external state
+- Are fully typed using TypeScript
+
+Here's an example of a simple component:
+
+```tsx
+interface GreetingProps {
+  name: string;
+}
+type GreetingOutput = string;
+
+const Greeting = gsx.Component<GreetingProps, GreetingOutput>(
+  async ({ name }) => {
+    return `Hello, ${name}!`;
+  },
+);
+```
+
+Components can also return other components using JSX:
+
+```tsx
+const AIGreeting = gsx.Component<GreetingProps, string>(async ({ name }) => {
+  return (
+    <ChatCompletion
+      model="gpt-4o-mini"
+      messages={[
+        {
+          role: "system",
+          content: "You are a friendly assistant that greets people warmly.",
+        },
+        { role: "user", content: `Write a greeting for ${name}.` },
+      ]}
+    />
+  );
+});
+```
+
+## JSX and Data Flow
+
+GenSX uses JSX to compose components into workflows. Data flows from parent to child through props allowing you to nest components and pass data between them. When components are combined, they will return the value of the innermost component.
```

Comment:
> Data flows from parent to child through props

I feel like this is not quite a precise as it needs to be. 

Yes props is the mechanism for receiving dependencies, but outputs and child function nesting is the mechanism for passing them. Don't want to overcomplicate here, but maybe we can make a small adjustment.

Reply:
I agree, I think we should be clear about the necessity of the child function, and how that works.

Reply:
Good feedback. Here's my latest draft of this: 
---
GenSX uses JSX to compose components into workflows. The output of a parent component can be passed to a child component through a child function. This pattern enables you to create chains of components where each step's output feeds into the next component's input.

When components are combined, they will return the value of the innermost component.